src/example-relay/src/graphql/locations/mutations/AddLocation.js

```diff
@@ -0,0 +1,26 @@
+// @flow strict-local
+
+import { GraphQLNonNull } from 'graphql';
+
+import AddLocationInput from '../types/input/AddLocationInput';
+import AddLocationOrError, { ValidLocationResponse } from '../types/output/AddLocationOrError';
+import type { LocationInput } from '../locations';
+
+type Args = {
+  +location: LocationInput,
+  ...
+};
+export default {
+  type: AddLocationOrError,
+  description: 'Add a new location',
+  args: {
+    location: { type: GraphQLNonNull(AddLocationInput) },
+  },
+  resolve: (_: mixed, { location }: Args) => {
+    if (location.name === '' || location.name == null) {
```

Comment:
No need for this if you make those param non null. Maybe duplicity check would be good here.

Reply:
We would still need to check for empty string. If you by duplicity mean that it is already not in the location.json, I think it is a little outside of the scope of this example.

Reply:
I'd agree it's out of scope of this PR but I think it's OK for the example to have example of a backend validation.

Reply:
Sure, why not let’s add it later 👍